### PR TITLE
MWPW-155099 - renew banner

### PR DESCRIPTION
**Pull request by draganatrajkovic**
**Resolves:** [MWPW-155099](https://jira.corp.adobe.com/browse/MWPW-155099)

- In order to test this, _**accountExpiration**_ in the partner_data cookie must be set
- if the paths for different banners are set in metdata.xlsx, they will be used; otherwise the default path will be /edsdme/partners-shared/fragments/${banner-type} with a locale prefix if the country is not the US
- currently, there are three types of banners: to recover a suspended account (active 90 days after expiration), for an account which expires in the next 24 hours, and for an account which expires in the next 30 days
- mobile design is coverd

**LINKS:**

**US page:**
**Before:** https://stage--dme-partners--adobecom.hlx.page/channelpartners/drafts/dragana/announcements
**After:** https://mwpw-155099-renew-banner--dme-partners--adobecom.hlx.page/channelpartners/drafts/dragana/announcements

Example:
Screenshot 2024-08-05 at 16 21 38

**DE page:**
**Before:** https://stage--dme-partners--adobecom.hlx.page/de/channelpartners/drafts/dragana/announcements
**After:** https://mwpw-155099-renew-banner--dme-partners--adobecom.hlx.page/de/channelpartners/drafts/dragana/announcements

Example:
Screenshot 2024-08-05 at 16 26 59

### CODE REVIEW

File: edsdme/scripts/utils.js [hunk 1]
```diff
@@ -150,6 +150,73 @@ export function redirectLoggedinPartner() {
...
+    if (differenceInDaysRounded === 0) {
+      metadataKey = 'banner-account-expires-today';
+    } else if (differenceInDays < 30) {
+      metadataKey = 'banner-account-expires';
+      daysNum = differenceInDaysRounded;
+    } else {
+      return;
+    }
+  } else if (differenceInMilliseconds < 0) {
+    if (differenceInDays < 90) {
```
Comment (Ben-Zahler) [inline edsdme/scripts/utils.js:178]: I guess this is something to be properly groomed: after the 90 days, the banner will disappear? does not really make sense, but the user should not be able to sign after 90 days of being suspended

Comment (draganatrajkovic) [reply edsdme/scripts/utils.js:178]: @Ben-Zahler Right, after 90 days banner will disappear

**Review by Ben-Zahler — APPROVED** [review]

**Review by cod17828 — APPROVED** [review]

**Review by dejanpreradovic — APPROVED** [review]

File: edsdme/scripts/utils.js [hunk 2]
```diff
@@ -150,6 +150,73 @@ export function redirectLoggedinPartner() {
   window.location.assign(target);
 }
 
+export async function getRenewBanner() {
+  const programType = getCurrentProgramType();
+  const accountExpiration = getPartnerDataCookieValue(programType, 'accountexpiration');
```
Comment (zagi25) [inline edsdme/scripts/utils.js:155]: This value is returned from runtime as  `accountAnniversary (accountanniversary)`
```
"partnerData": {
        "CPP": {
            "accountAnniversary": "2025-07-17T00:00:00.000Z",
            "company": "Yugo Stage CPP  Gold United Kingdom",
            "firstName": "Yugo Stage CPP",
            "lastName": "Gold United Kingdom",
            "level": "Gold",
            "permissionRegion": "United Kingdom,Europe West",
            "status": "MEMBER"
        }
    },
```

Comment (draganatrajkovic) [reply edsdme/scripts/utils.js:155]: @zagi25 Thank you, will change the name

File: edsdme/scripts/utils.js [hunk 3]
```diff
@@ -150,6 +150,73 @@ export function redirectLoggedinPartner() {
...
+      daysNum = 90 - differenceInDaysRounded;
+    } else {
+      return;
+    }
+  }
+
+  const config = getConfig();
+  const [, locale] = config.locale.ietf.toLowerCase().split('-');
+  const defaultPath = locale === 'us' ? `/edsdme/partners-shared/fragments/${metadataKey}` : `/${locale}/edsdme/partners-shared/fragments/${metadataKey}`;
+  const path = getMetadataContent(metadataKey) ? getMetadataContent(metadataKey) : defaultPath;
```
Comment (zagi25) [inline edsdme/scripts/utils.js:189]: `getMetadataContent` returns `undefined` if there is no meta tag with a given name, so we can shorten this to `getMetadataContent(metadataKey) ?? defaultPath`

File: edsdme/scripts/utils.js [hunk 4]
```diff
@@ -150,6 +150,73 @@ export function redirectLoggedinPartner() {
...
+      metadataKey = 'banner-account-suspended';
+      daysNum = 90 - differenceInDaysRounded;
+    } else {
+      return;
+    }
+  }
+
+  const config = getConfig();
+  const [, locale] = config.locale.ietf.toLowerCase().split('-');
+  const defaultPath = locale === 'us' ? `/edsdme/partners-shared/fragments/${metadataKey}` : `/${locale}/edsdme/partners-shared/fragments/${metadataKey}`;
```
Comment (zagi25) [inline edsdme/scripts/utils.js:188]: `config.locale` contains property `prefix` (eg. for 'us' is '' and for 'de' is '/de'), this can be used to construct url: `${locale.prefix}/edsdme/partners-shared/fragments/${metadataKey}`

File: edsdme/scripts/utils.js [hunk 5]
```diff
@@ -50,7 +50,7 @@ export async function useMiloSample() {
 
 const miloLibs = setLibs('/libs');
 
-const { createTag, localizeLink, getConfig } = await import(`${miloLibs}/utils/utils.js`);
+const { createTag, localizeLink, getConfig, loadBlock } = await import(`${miloLibs}/utils/utils.js`);
```
Comment (zagi25) [inline edsdme/scripts/utils.js:53]: In performance pr this is removed from `utils.js`, so you can either import necessary MILO function in `scripts.js` and passed them as params to `getRenewBanner`, or import functions inside `getRenewBanner` function.

**Review by zagi25 — APPROVED** [review]

File: edsdme/scripts/utils.js (hunk 4 above)
Comment (draganatrajkovic) [reply edsdme/scripts/utils.js]: Great suggestion, thank you!
